Examples/Examples/All Examples/SwiftUIExample.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+import MapboxMaps
+import Turf
+import SwiftUI
+
+/// Example of how a SwiftUI map view can be used within a typical `ContentView`.
+public struct ContentView: View {
+
+    public var body: some View {
+        SwiftUIMapView(resourceOptions: ResourceOptions(accessToken: AccountManager.shared.accessToken!))
+            .styleURL(.dark)
+            .zoomLevel(14.0)
+            .centerCoordinate(CLLocationCoordinate2D(latitude: 20.6905, longitude: -88.2024))
+            .on(.mapLoadingFinished) { ( mapView ) in
+                let pointAnnotation = PointAnnotation(coordinate: mapView.centerCoordinate)
+                mapView.annotationManager.addAnnotation(pointAnnotation)
+            }
+    }
+}
+
+/// To render a `MapView` as a SwiftUI view, configure a new
+/// struct that implements the required `UIViewRepresentable` protocol.
+public struct SwiftUIMapView: UIViewRepresentable {
+
+    let resourceOptions: ResourceOptions
+    internal let observerConcrete: ObserverConcrete // Manages the observation of map events
+    internal var mapView: MapView
+
+    init(resourceOptions: ResourceOptions) {
+        self.resourceOptions = resourceOptions
+        self.mapView = MapView(with: .zero, resourceOptions: resourceOptions)
```

Comment:
Based on a brief reading of the UIViewRepresentable docs, I think we should probably _not_ be creating MapView until `makeUIView(context:)` and also not holding a reference to it.

I think what we'd want to do instead is to have SwiftUIMapView hold the various configurable values, and only apply them to the map view in `updateUIView(_:context:)`

Reply:
This approach allows SwiftUI to own and manage the MapView instance and avoids the possibility of bugs due to multiple SwiftUIMapView values having a reference to a single MapView instance.